scripts/env.d/devenv_impl

```diff
@@ -17,6 +17,14 @@ nameremove() {
   eval "export $1=$(echo -n ${!1} | awk -v RS=: -v ORS=: -v var="$2" '$0 != var' | sed 's/:*$//')"
 }
 
+# Necessary evil
+if [[ -f "/etc/os-release" ]]; then
+    str=$(grep -o 'ubuntu' /etc/os-release | head -1)
+    if [[ ! -z str ]]; then
+      sudo apt install build-essential -y
```

Comment:
I am not sure if it is a good idea to embed `sudo` in the tool.

1. If I understand correctly, `devenv` could be use (and probably supposed to be used) without `sudo` (please correct me if I am wrong, thanks @j8xixo12  @chestercheng !), because we have already built tools from source and install it in the specific `flavor`.
2. `build-essential` is essential for some tools, but maybe not for the tool we are aiming at.

What do you think if it offers hints only instead?

Reply:
I am sure using sudo is a bad idea.  Do not require root permission in devenv.  It will render the tool useless in one of its most useful cases: user-mode thirdparty managment.

Reply:
Got it, I'll avoid using the system permission and find another way to solve the "make command not found" problem.